pkg/pulumiyaml/analyser.go

```diff
@@ -361,10 +361,23 @@ func (tc *typeCache) isAssignable(fromExpr ast.Expr, to schema.Type) *notAssigna
 		return okIfAssignable(hasValidEnumValue(fromExpr, to.Elements))
 	case *schema.ObjectType:
 		// We implement structural typing for objects.
-		from, ok := from.(*schema.ObjectType)
-		if !ok {
+		switch f := from.(type) {
```

Comment:
Previously, we implicitly converted objects into maps, but we had no facility to interpret maps as objects. Because our `object` type matches objects with no structural information about its values, we now have to handle this case.